gloss/crt0.S

```diff
@@ -293,6 +293,7 @@ secondary_main:
 /* This shim allows main() to be passed a set of arguments that can satisfy the
  * requirements of the C API. */
 .section .rodata.libgloss.start
+.align 4
```

Comment:
I would suggest align to 8 rather than 4, consider RV64 load a XLEN data, that will use `ld` which mean require 8 byte align.

Reply:
Agreed. 

At a higher level, I'm never quite sure where is the appropriate place to enforce alignment. See, e.g., 
https://github.com/sifive/ldscript-generator/blob/master/templates/base.lds#L184-L196
https://github.com/sifive/ldscript-generator/blob/master/templates/base.lds#L269-L294
and this comment:
https://github.com/sifive/ldscript-generator/blob/master/templates/base.lds#L263-L266

Reply:
If I'm not mistaken:
* `.align 4` means align on 16-byte boundary
* `.align 8` means align on 256-byte boundary

https://github.com/riscv/riscv-asm-manual/blob/master/riscv-asm.md#pseudo-ops

Reply:
Right... on X86, `.align 4` means "align to 4-byte boundary, but on ARM and RISC-V, `.align 4` means "align to 2**4-byte boundary". Personally I prefer to use `.balign` to avoid confusion.

Reply:
Very true: `.align` should be avoided, as it is highly error-prone.